enclave/core/sgx/exit.S

```diff
@@ -74,24 +89,28 @@ oe_asm_exit:
     lfence
 
     // Clear the oe_sgx_td_t.last_sp field (force oe_enter to calculate stack pointer)
-    movq $0, %fs:td_last_sp
+    movq $0, td_last_sp(%r9)
 
 .clear_enclave_registers:
 
-    // Clear these so information will not be leaked to host
+    // Clear registers so information will not be leaked to host
+    // Retain %r9 which is used to reference the td structure. It will be
+    // cleared immediately before EEXIT.
+    push %r9
```

Comment:
This seems like a workaround for oe_cleanup_register's current behavior. That function can be modified instead.

Reply:
Good point. I could use r11 in both enter and exit and then just change the cleanup code to not clear r11.

Reply:
Done